plugins/module_utils/redfish_utils.py

```diff
@@ -903,16 +903,13 @@ def get_volume_inventory(self, systems_uri):
                     if response['ret'] is False:
                         return response
                     data = response['data']
-                    controller_name = 'Controller 1'
-                    if 'StorageControllers' in data:
-                        sc = data['StorageControllers']
-                        if sc:
-                            if 'Name' in sc[0]:
-                                controller_name = sc[0]['Name']
-                            else:
-                                sc_id = sc[0].get('Id', '1')
-                                controller_name = 'Controller %s' % sc_id
+                    if 'Name' in data:
```

Comment:
This logic isn't correct; this pulls the name of the storage subsystem rather than of the storage controller.

Inside the Storage resource is a "StorageControllers" property; this property contains the set of storage controllers in the storage subsystem, which is what this `controller_list` is intended to contain.

Reply:
Actually this is under a for loop which is iterating through each controller.
```
               for controller in data[u'Members']:
                    controller_list.append(controller[u'@odata.id'])
                for c in controller_list:
                    uri = self.root_uri + c
                    response = self.get_request(uri)
                    if response['ret'] is False:
                        return response
                    data = response['data']
                    controller_name = 'Controller 1'
                    if 'StorageControllers' in data:
                        sc = data['StorageControllers']
                        if sc:
                            if 'Name' in sc[0]:
                                controller_name = sc[0]['Name']
                            else:
                                sc_id = sc[0].get('Id', '1')
                                controller_name = 'Controller %s' % sc_id
                    if 'Name' in data:
```